modules/vpc/main.tf

```diff
@@ -20,25 +13,25 @@ resource "aws_vpc" "main" {
 
 #Subnets
 resource "aws_subnet" "main-public" {
-  count = 2
+  count = var.public_number
 
   vpc_id                  = aws_vpc.main.id
-  cidr_block              = local.public_cidr_block[count.index]
+  cidr_block              = var.public_cidr_block[count.index]
   map_public_ip_on_launch = "true"
-  availability_zone       = local.az[count.index]
+  availability_zone       = var.az[count.index]
 
   tags = {
     Name = "main-public-${count.index + 1}"
   }
 }
 
 resource "aws_subnet" "main-private" {
-  count = 2
+  count = var.private_number
```

Comment:
same as my previous comment and for any other count in this module

Reply:
fixed